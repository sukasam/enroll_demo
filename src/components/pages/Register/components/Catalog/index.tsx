/** @jsxImportSource @emotion/react */
import { useUser } from "Contexts/UserContext";
import { useState } from "react";
import CustomizerModal from "./components/CustomizerModal";
import NotReadySection from "./components/NotReadySection";
import PackSection from "./components/PackSection";

export default function Home(): JSX.Element {
    const [customizerModalID, setCustomizerModalID] = useState<number | null>(
        null
    );

    const { enrollerId } = useUser();

    return (
        <>
            <PackSection />
            <NotReadySection enrollerId={enrollerId} />
            {customizerModalID !== null && (
                <CustomizerModal
                    customizerModalID={customizerModalID}
                    setCustomizerModalID={setCustomizerModalID}
                />
            )}
        </>
    );
}
