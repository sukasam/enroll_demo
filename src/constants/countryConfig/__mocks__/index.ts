export const useCountryConfig = jest.fn().mockReturnValue({
    marketExceptions: {
        showOnboardingGuide: true,
        hideUnicityPhone: false,
        thankYouImage: "/img/custom-image.jpg"
    }
});

export const getSuppotedCountries = jest.fn();
export const getUnsupportedCountries = jest.fn();
export const getCountries = jest.fn();
export const getCountryConfig = jest.fn();
export const isSupported = jest.fn();
export const getDefaultLanguage = jest.fn();
export const countryConfigs = [];
