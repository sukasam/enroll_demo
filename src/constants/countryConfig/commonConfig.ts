import { Currency, optInNotificationTypes } from "./enums";
import {
    MarketExceptions,
    PaymentLogo,
    PaymentOptions,
    ProductConfig,
    TemplateSupportedCountryConfig
} from "./types";

const productConfig: ProductConfig = {
    recommendedPack: 0,
    packs: [
        {
            isDigital: false,
            buttonName: "catalog_get_business_builder",
            kitDescription: "business_builder_kit_description",
            mainFeatures: [
                "catalog_modal_business_builder_main_features_1",
                "catalog_modal_business_builder_main_features_2",
                "catalog_modal_business_builder_main_features_3",
                "catalog_modal_business_builder_main_features_4",
                "catalog_modal_business_builder_main_features_5",
                "catalog_modal_business_builder_main_features_6",
                "catalog_modal_business_builder_main_features_7",
                "catalog_modal_business_builder_main_features_8"
            ],
            accentColor: "#FAA977",
            backgroundColor: "#FBF7F1",
            perks: [
                {
                    icon: "/svg/page-with-checkmark.svg",
                    perk: "business_builder_perk_1"
                },
                {
                    icon: "/svg/phone-with-checkmark.svg",
                    perk: "business_builder_perk_2"
                },
                {
                    icon: "/svg/add-people.svg",
                    perk: "business_builder_perk_3"
                }
            ],
            extras: [
                {
                    icon: "/svg/dollar-checkmark-circle.svg",
                    title: "business_builder_extra_1_title",
                    description: "business_builder_extra_1_description"
                }
            ],
            products: [
                {
                    image: "/img/unimate-lemon-3.png",
                    title: "business_builder_product_1_title",
                    description: "business_builder_product_1_description"
                },
                {
                    image: "/img/balance-3.png",
                    title: "business_builder_product_2_title",
                    description: "business_builder_product_2_description"
                }
            ],
            annotations: "business_builder_annotations",
            showPaymentType: true
        },
        {
            name: "basic_name",
            buttonName: "catalog_basic_distributor_only",
            description: "basic_description",
            isDigital: true,
            showPaymentType: true
        }
    ]
};

const paymentOptionsConfig: PaymentOptions = {
    hasPayPal: false,
    hasCash: false,
    hasBankWire: false,
    hasCreditCard: false,
    hasSavedCreditCard: false
};

const marketExceptions: MarketExceptions = {
    zipLength: [5, 9],
    taxIdPattern: undefined,
    hasSecondaryTaxId: false,
    taxInclusiveBackEnd: false,
    taxInclusiveFrontEnd: false,
    hiddenShippingProvince: true,
    useQuotesToValidateAddress: true,
    hasCommissionPaymentOptions: true,
    showOnboardingGuide: true,
    showVisaAgreement: false,
    paymentOptions: paymentOptionsConfig,
    useWorldPay: false,
    usePayPal3ds: false,
    notificationType: optInNotificationTypes.SMS,
    hiddenDecimal: false,
    updateMainAddress: true,
    formCreateAccount: [
        "firstName",
        "lastName",
        "phoneNumber",
        "email",
        "password"
    ],
    formUpdateAccount: ["firstName", "lastName", "phoneNumber", "email"]
};

const commonConfig: TemplateSupportedCountryConfig = {
    dateFormat: "MDY",
    languages: [
        {
            code: "en",
            name: "English",
            default: true
        }
    ],
    currency: Currency.USD,
    alwaysShowGuide: false,
    paymentLogos: [
        PaymentLogo.Visa,
        PaymentLogo.Mastercard,
        PaymentLogo.Amex,
        PaymentLogo.Discover,
        PaymentLogo.JCB
    ],
    productConfig,
    marketExceptions
};

export default commonConfig;
