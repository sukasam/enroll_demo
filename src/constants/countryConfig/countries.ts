import AD from "./countries/AD";
import AE from "./countries/AE";
import AF from "./countries/AF";
import AG from "./countries/AG";
import AI from "./countries/AI";
import AL from "./countries/AL";
import AM from "./countries/AM";
import AO from "./countries/AO";
import AQ from "./countries/AQ";
import AR from "./countries/AR";
import AS from "./countries/AS";
import AT from "./countries/AT";
import AU from "./countries/AU";
import AW from "./countries/AW";
import AX from "./countries/AX";
import AZ from "./countries/AZ";
import BA from "./countries/BA";
import BB from "./countries/BB";
import BD from "./countries/BD";
import BE from "./countries/BE";
import BF from "./countries/BF";
import BG from "./countries/BG";
import BH from "./countries/BH";
import BI from "./countries/BI";
import BJ from "./countries/BJ";
import BL from "./countries/BL";
import BM from "./countries/BM";
import BN from "./countries/BN";
import BO from "./countries/BO";
import BQ from "./countries/BQ";
import BR from "./countries/BR";
import BT from "./countries/BT";
import BV from "./countries/BV";
import BW from "./countries/BW";
import BY from "./countries/BY";
import BZ from "./countries/BZ";
import CA from "./countries/CA";
import CC from "./countries/CC";
import CD from "./countries/CD";
import CF from "./countries/CF";
import CG from "./countries/CG";
import CH from "./countries/CH";
import CI from "./countries/CI";
import CK from "./countries/CK";
import CL from "./countries/CL";
import CM from "./countries/CM";
import CN from "./countries/CN";
import CO from "./countries/CO";
import CR from "./countries/CR";
import CU from "./countries/CU";
import CV from "./countries/CV";
import CW from "./countries/CW";
import CX from "./countries/CX";
import CY from "./countries/CY";
import CZ from "./countries/CZ";
import DE from "./countries/DE";
import DJ from "./countries/DJ";
import DK from "./countries/DK";
import DM from "./countries/DM";
import DO from "./countries/DO";
import DZ from "./countries/DZ";
import EC from "./countries/EC";
import EE from "./countries/EE";
import EG from "./countries/EG";
import EH from "./countries/EH";
import ER from "./countries/ER";
import ES from "./countries/ES";
import ET from "./countries/ET";
import FI from "./countries/FI";
import FJ from "./countries/FJ";
import FK from "./countries/FK";
import FM from "./countries/FM";
import FO from "./countries/FO";
import FR from "./countries/FR";
import GA from "./countries/GA";
import GB from "./countries/GB";
import GC from "./countries/GC";
import GD from "./countries/GD";
import GE from "./countries/GE";
import GF from "./countries/GF";
import GG from "./countries/GG";
import GH from "./countries/GH";
import GI from "./countries/GI";
import GL from "./countries/GL";
import GM from "./countries/GM";
import GN from "./countries/GN";
import GP from "./countries/GP";
import GQ from "./countries/GQ";
import GR from "./countries/GR";
import GS from "./countries/GS";
import GT from "./countries/GT";
import GU from "./countries/GU";
import GW from "./countries/GW";
import GY from "./countries/GY";
import HK from "./countries/HK";
import HM from "./countries/HM";
import HN from "./countries/HN";
import HR from "./countries/HR";
import HT from "./countries/HT";
import HU from "./countries/HU";
import IE from "./countries/IE";
import IL from "./countries/IL";
import IM from "./countries/IM";
import IN from "./countries/IN";
import IO from "./countries/IO";
import IQ from "./countries/IQ";
import IR from "./countries/IR";
import IS from "./countries/IS";
import IT from "./countries/IT";
import JE from "./countries/JE";
import JO from "./countries/JO";
import JP from "./countries/JP";
import KE from "./countries/KE";
import KG from "./countries/KG";
import KH from "./countries/KH";
import KI from "./countries/KI";
import KM from "./countries/KM";
import KN from "./countries/KN";
import KP from "./countries/KP";
import KR from "./countries/KR";
import KW from "./countries/KW";
import KY from "./countries/KY";
import KZ from "./countries/KZ";
import LA from "./countries/LA";
import LB from "./countries/LB";
import LC from "./countries/LC";
import LI from "./countries/LI";
import LK from "./countries/LK";
import LR from "./countries/LR";
import LS from "./countries/LS";
import LT from "./countries/LT";
import LU from "./countries/LU";
import LV from "./countries/LV";
import LY from "./countries/LY";
import MA from "./countries/MA";
import MC from "./countries/MC";
import MD from "./countries/MD";
import ME from "./countries/ME";
import MF from "./countries/MF";
import MG from "./countries/MG";
import MH from "./countries/MH";
import MK from "./countries/MK";
import ML from "./countries/ML";
import MN from "./countries/MN";
import MO from "./countries/MO";
import MP from "./countries/MP";
import MQ from "./countries/MQ";
import MR from "./countries/MR";
import MS from "./countries/MS";
import MT from "./countries/MT";
import MU from "./countries/MU";
import MV from "./countries/MV";
import MW from "./countries/MW";
import MX from "./countries/MX";
import MY from "./countries/MY";
import MZ from "./countries/MZ";
import NA from "./countries/NA";
import NC from "./countries/NC";
import NE from "./countries/NE";
import NF from "./countries/NF";
import NG from "./countries/NG";
import NI from "./countries/NI";
import NL from "./countries/NL";
import NO from "./countries/NO";
import NP from "./countries/NP";
import NR from "./countries/NR";
import NU from "./countries/NU";
import NZ from "./countries/NZ";
import OM from "./countries/OM";
import PA from "./countries/PA";
import PE from "./countries/PE";
import PF from "./countries/PF";
import PG from "./countries/PG";
import PH from "./countries/PH";
import PK from "./countries/PK";
import PL from "./countries/PL";
import PM from "./countries/PM";
import PN from "./countries/PN";
import PR from "./countries/PR";
import PS from "./countries/PS";
import PT from "./countries/PT";
import PW from "./countries/PW";
import PY from "./countries/PY";
import QA from "./countries/QA";
import RE from "./countries/RE";
import RO from "./countries/RO";
import RS from "./countries/RS";
import RU from "./countries/RU";
import RW from "./countries/RW";
import SA from "./countries/SA";
import SB from "./countries/SB";
import SC from "./countries/SC";
import SD from "./countries/SD";
import SE from "./countries/SE";
import SG from "./countries/SG";
import SH from "./countries/SH";
import SI from "./countries/SI";
import SJ from "./countries/SJ";
import SK from "./countries/SK";
import SL from "./countries/SL";
import SM from "./countries/SM";
import SN from "./countries/SN";
import SO from "./countries/SO";
import SR from "./countries/SR";
import SS from "./countries/SS";
import ST from "./countries/ST";
import SV from "./countries/SV";
import SX from "./countries/SX";
import SY from "./countries/SY";
import SZ from "./countries/SZ";
import TC from "./countries/TC";
import TD from "./countries/TD";
import TF from "./countries/TF";
import TG from "./countries/TG";
import TH from "./countries/TH";
import TJ from "./countries/TJ";
import TK from "./countries/TK";
import TL from "./countries/TL";
import TM from "./countries/TM";
import TN from "./countries/TN";
import TO from "./countries/TO";
import TR from "./countries/TR";
import TT from "./countries/TT";
import TV from "./countries/TV";
import TW from "./countries/TW";
import TZ from "./countries/TZ";
import UG from "./countries/UG";
import UM from "./countries/UM";
import US from "./countries/US";
import UY from "./countries/UY";
import UZ from "./countries/UZ";
import VA from "./countries/VA";
import VC from "./countries/VC";
import VE from "./countries/VE";
import VG from "./countries/VG";
import VI from "./countries/VI";
import VN from "./countries/VN";
import VU from "./countries/VU";
import WF from "./countries/WF";
import WS from "./countries/WS";
import YE from "./countries/YE";
import YT from "./countries/YT";
import ZA from "./countries/ZA";
import ZM from "./countries/ZM";
import ZW from "./countries/ZW";
import { CountryConfig } from "./types";

export default [
    US,
    CA,
    PR,
    GB,
    AE,
    AI,
    AT,
    AU,
    BB,
    BE,
    BH,
    BQ,
    CH,
    CO,
    CW,
    CZ,
    DE,
    DK,
    DM,
    DO,
    DZ,
    EG,
    ES,
    FR,
    GD,
    HU,
    IE,
    IN,
    IT,
    JO,
    JP,
    KE,
    KR,
    KW,
    LB,
    LU,
    MX,
    NL,
    NO,
    NZ,
    OM,
    PH,
    PL,
    SA,
    SE,
    SG,
    TC,
    TH,
    TR,
    TT,
    VI,
    MA,
    LA,
    MY,
    TW,
    VN,
    HK,
    KH,
    AF,
    AX,
    AL,
    AS,
    AD,
    AO,
    AQ,
    AG,
    AR,
    AM,
    AW,
    AZ,
    BD,
    BY,
    BZ,
    BJ,
    BM,
    BT,
    BO,
    BA,
    BW,
    BV,
    BR,
    IO,
    BN,
    BG,
    BF,
    BI,
    CM,
    CV,
    KY,
    CF,
    TD,
    CL,
    CN,
    CX,
    CC,
    KM,
    CK,
    CR,
    CI,
    HR,
    CU,
    CY,
    CD,
    DJ,
    EC,
    SV,
    GQ,
    ER,
    EE,
    ET,
    FK,
    FO,
    FJ,
    FI,
    GF,
    PF,
    TF,
    GA,
    GE,
    GH,
    GI,
    GR,
    GL,
    GP,
    GU,
    GT,
    GG,
    GN,
    GW,
    GC,
    GY,
    HT,
    HM,
    HN,
    IS,
    IR,
    IQ,
    IM,
    IL,
    JE,
    KZ,
    KI,
    KG,
    LV,
    LS,
    LR,
    LY,
    LI,
    LT,
    MO,
    MG,
    MW,
    MV,
    ML,
    MT,
    MH,
    MQ,
    MR,
    MU,
    YT,
    FM,
    MD,
    MC,
    MN,
    ME,
    MS,
    MZ,
    NA,
    NR,
    NP,
    NC,
    NI,
    NE,
    NG,
    NU,
    NF,
    KP,
    MP,
    PK,
    PW,
    PA,
    PG,
    PY,
    PE,
    PN,
    PT,
    QA,
    MK,
    CG,
    RE,
    RO,
    RU,
    RW,
    BL,
    SH,
    KN,
    LC,
    MF,
    PM,
    VC,
    WS,
    SM,
    ST,
    SN,
    RS,
    SC,
    SL,
    SX,
    SK,
    SI,
    SB,
    SO,
    ZA,
    GS,
    SS,
    LK,
    PS,
    SD,
    SR,
    SJ,
    SZ,
    SY,
    TJ,
    TZ,
    GM,
    TL,
    TG,
    TK,
    TO,
    TN,
    TM,
    TV,
    UG,
    UM,
    UY,
    UZ,
    VU,
    VA,
    VE,
    VG,
    WF,
    EH,
    YE,
    ZM,
    ZW
] as CountryConfig[];
