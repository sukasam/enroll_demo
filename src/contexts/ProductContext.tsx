import {
    ReactElement,
    ReactNode,
    createContext,
    useCallback,
    useContext,
    useEffect,
    useMemo,
    useState
} from "react";
import noPurchaseVariantData from "./data/NoPurchaseProductData";
import useProductCalculations from "./hooks/ProductCalculations";
import {
    Product,
    ProductContextType,
    Variant
} from "./types/ProductContextTypes";

const defaultProductContext: ProductContextType = {
    selectedSku: null,
    setSelectedSku: () => null,
    shoppingCart: [],
    setShoppingCart: () => null,
    products: [],
    setProducts: () => null,
    mappedProducts: [],
    setMappedProducts: () => null,
    href: null,
    setHref: () => null,
    availableDigitalGuideVariant: null,
    digitalVariants: [],
    ufgVariants: [],
    recommendedPack: null,
    digitalPack: null,
    feelGreatPack: null,
    noPurchaseVariantData: []
};

export const ProductContext = createContext<ProductContextType>(
    defaultProductContext
);

ProductContext.displayName = "Product Context";

export function ProductProvider({
    children,
    testValues
}: {
    children: ReactNode;
    testValues?: Record<string, string>;
}): ReactElement {
    const [selectedSku, setSelectedSku] = useState<string | null>(
        defaultProductContext.selectedSku
    );
    const [shoppingCart, setShoppingCartState] = useState<Variant[]>(
        defaultProductContext.shoppingCart
    );
    const [products, setProducts] = useState<Product[]>(
        defaultProductContext.products
    );
    const [mappedProducts, setMappedProducts] = useState<Variant[]>(
        defaultProductContext.mappedProducts
    );
    const [href, setHref] = useState<string | null>(defaultProductContext.href);

    const {
        availableDigitalGuideVariant,
        ufgVariants,
        digitalVariants,
        digitalPack,
        feelGreatPack,
        recommendedPack
    } = useProductCalculations(shoppingCart, products);

    const setShoppingCart = useCallback(
        (newCart: Variant[] | null): void => {
            if (newCart && newCart.length === 0) {
                setShoppingCartState([]);
                return;
            }

            let cart = newCart || shoppingCart;
            const hasDigitalGuide = cart?.find(variant => variant.isDigital);

            if (availableDigitalGuideVariant && hasDigitalGuide) {
                cart = cart?.filter(variant => !variant.isDigital);
                cart.push(availableDigitalGuideVariant);
                setShoppingCartState(cart);
                return;
            }

            if (
                cart &&
                cart.length > 0 &&
                !hasDigitalGuide &&
                availableDigitalGuideVariant
            ) {
                cart.push(availableDigitalGuideVariant);
            }
            setShoppingCartState(cart);
        },
        [shoppingCart, availableDigitalGuideVariant, setShoppingCartState]
    );

    useEffect((): void => {
        setShoppingCart(null);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [availableDigitalGuideVariant]);

    const loadCachedProductData = (): void => {
        const cachedData = sessionStorage.getItem("productContext");
        if (cachedData) {
            const parsedData: Pick<
                ProductContextType,
                "selectedSku" | "products" | "shoppingCart" | "mappedProducts"
            > = JSON.parse(cachedData);
            setSelectedSku(parsedData.selectedSku);
            setShoppingCart(parsedData.shoppingCart);
            setProducts(parsedData.products);
            setMappedProducts(parsedData.mappedProducts);
        }
    };

    useEffect(() => {
        loadCachedProductData();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const updateCachedProductData = useCallback((): void => {
        const dataToCache = {
            selectedSku,
            products,
            shoppingCart,
            mappedProducts
        };
        sessionStorage.setItem("productContext", JSON.stringify(dataToCache));
    }, [selectedSku, products, shoppingCart, mappedProducts]);

    useEffect(() => {
        updateCachedProductData();
    }, [updateCachedProductData]);

    const value = useMemo(
        () => ({
            selectedSku,
            setSelectedSku,
            shoppingCart,
            setShoppingCart,
            products,
            setProducts,
            mappedProducts,
            setMappedProducts,
            href,
            setHref,
            availableDigitalGuideVariant,
            digitalVariants,
            ufgVariants,
            digitalPack,
            feelGreatPack,
            recommendedPack,
            noPurchaseVariantData
        }),
        [
            selectedSku,
            products,
            shoppingCart,
            setShoppingCart,
            mappedProducts,
            href,
            availableDigitalGuideVariant,
            digitalVariants,
            ufgVariants,
            digitalPack,
            feelGreatPack,
            recommendedPack,
            noPurchaseVariantData
        ]
    );

    const providerValue = useMemo(
        () => ({
            ...value,
            ...testValues
        }),
        [value, testValues]
    );

    return (
        <ProductContext.Provider value={providerValue}>
            {children}
        </ProductContext.Provider>
    );
}

export const useProducts = (): ProductContextType => useContext(ProductContext);
