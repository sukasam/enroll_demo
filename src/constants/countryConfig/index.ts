import { useTranslations } from "Contexts/translation";
import { useMemo } from "react";
import countries from "./countries";
import { Alpha2 } from "./enums";
import {
    CountryConfig,
    Locale,
    SupportedCountryConfig,
    UnsupportedCountryConfig
} from "./types";

// This makes importing less confusing
export { countries as countryConfigs };

export function getSuppotedCountries(): SupportedCountryConfig[] {
    return countries.filter<SupportedCountryConfig>(
        (config): config is SupportedCountryConfig => config.isSupported
    );
}

export function getUnsupportedCountries(): UnsupportedCountryConfig[] {
    return countries.filter(
        (config): config is UnsupportedCountryConfig => !config.isSupported
    );
}

export function getCountries(supported: boolean): CountryConfig[] {
    if (supported) {
        return getSuppotedCountries();
    }

    return countries;
}

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: true
): SupportedCountryConfig | undefined;

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: false
): CountryConfig | undefined;

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: boolean
): CountryConfig | undefined {
    if (supportedOnly) {
        return getSuppotedCountries().find(
            config => config.alpha2 === countryCode
        );
    }

    return countries.find(config => config.alpha2 === countryCode);
}

export function isSupported(
    config: CountryConfig
): config is SupportedCountryConfig {
    return config.isSupported;
}

export function useCountryConfig(): SupportedCountryConfig | undefined {
    const { country } = useTranslations();
    const alpha2 = country as Alpha2;
    const config = useMemo(() => getCountryConfig(alpha2, true), [alpha2]);
    return config;
}

export function getDefaultLanguage(alpha2: Alpha2): string {
    const config = getCountryConfig(alpha2, true);

    return config?.languages.find(lang => lang.default)?.code ?? "en";
}

export function getValidatedLocale(
    countryCode: string | null | Alpha2,
    language: string | null
): Locale {
    // Default to US if no country or invalid country
    const countryConfig =
        getCountryConfig(countryCode as Alpha2, true) ||
        getCountryConfig(Alpha2.US, true);

    // Get default language for the country
    const defaultLanguage = getDefaultLanguage(countryCode as Alpha2);

    // Validate the provided language against country's available languages
    const isValidLanguage =
        language &&
        countryConfig?.languages?.some(lang => lang.code === language);

    return {
        countryCode: countryConfig?.alpha2 || Alpha2.US,
        language: isValidLanguage ? language! : defaultLanguage
    };
}
