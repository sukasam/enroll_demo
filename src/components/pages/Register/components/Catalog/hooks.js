import { useCountryConfig } from "Constants/countryConfig";
import { useProducts } from "Contexts/ProductContext";
import { useTranslations } from "Contexts/translation";
import getProducts from "Hydra/getProducts";
import { useEffect } from "react";

export default function useLoadProducts() {
    const { setProducts } = useProducts();
    const { country, language } = useTranslations();
    const countryConfig = useCountryConfig();

    async function load() {
        if (countryConfig?.isNoPurchaseMarket) return;

        const catalog = await getProducts({
            alpha2: country,
            customerType: "Member",
            language
        });
        setProducts(catalog.catalogSlides);
    }

    useEffect(() => {
        load();
    }, [language, country]);
}
