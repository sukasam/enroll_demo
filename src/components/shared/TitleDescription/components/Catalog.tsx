/** @jsxImportSource @emotion/react */
import { useTheme } from "@mui/material";
import T from "Components/shared/Translate";
import { useProducts } from "Contexts/ProductContext";
import { Variant } from "Contexts/types/ProductContextTypes";
import { useMemo } from "react";
import styles from "./styles";

export default function Catalog(): JSX.Element | null {
    const theme = useTheme();
    const { shoppingCart } = useProducts();

    const selectedUFGProduct: Variant | null = useMemo(
        () =>
            shoppingCart?.find(
                product =>
                    product?.options && Object.keys(product.options).length > 0
            ) || null,
        [shoppingCart]
    );

    const selectedDigitalProduct: Variant | null = useMemo(
        () => shoppingCart?.find(product => product?.isDigital) || null,
        [shoppingCart]
    );

    const unimateFlavor: string = useMemo(
        () => selectedUFGProduct?.options?.unimate_flavour?.title || "n/a",
        [selectedUFGProduct]
    );

    const balanceFlavor: string = useMemo(
        () => selectedUFGProduct?.options?.balance_pack?.title || "n/a",
        [selectedUFGProduct]
    );

    if (selectedUFGProduct) {
        return (
            <div
                css={styles.catalogContainer(theme)}
                data-testid="catalog_unimate_flavor"
            >
                <div>
                    <T>catalog_unimate_flavor</T> {unimateFlavor}
                </div>
                <div>
                    <T>catalog_balance_flavor</T> {balanceFlavor}
                </div>
            </div>
        );
    }

    if (selectedDigitalProduct) {
        return (
            <div data-testid="catalog_digital_product">
                <T>catalog_basic_distributor_only</T>
            </div>
        );
    }

    return null;
}
