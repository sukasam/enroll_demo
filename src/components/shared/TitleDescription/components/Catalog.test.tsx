import { useCountryConfig } from "Constants/countryConfig";
import { useProducts } from "Contexts/ProductContext";
import { render } from "Shared/testUtils";
import Catalog from "./Catalog";

// Mock the necessary hooks and components
jest.mock("Contexts/ProductContext", () => ({
    useProducts: jest.fn()
}));

jest.mock("Constants/countryConfig", () => ({
    useCountryConfig: jest.fn()
}));

jest.mock(
    "Components/shared/Translate",
    () =>
        function Translate({
            children
        }: {
            children: React.ReactNode;
        }): JSX.Element {
            return <span>{children}</span>;
        }
);

describe("Catalog", () => {
    beforeEach(() => {
        (useProducts as jest.Mock).mockReturnValue({
            shoppingCart: []
        });

        (useCountryConfig as jest.Mock).mockReturnValue({});
    });

    it("renders correctly with selected UFG product", () => {
        (useProducts as jest.Mock).mockReturnValue({
            shoppingCart: [
                {
                    options: {
                        unimate_flavour: { title: "Lemon" },
                        balance_pack: { title: "Orange" }
                    }
                }
            ]
        });

        const { asFragment } = render(<Catalog />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders correctly with selected digital product", () => {
        (useProducts as jest.Mock).mockReturnValue({
            shoppingCart: [
                {
                    isDigital: true
                }
            ]
        });

        const { asFragment } = render(<Catalog />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders correctly with no purchase market", () => {
        (useCountryConfig as jest.Mock).mockReturnValue({
            isNoPurchaseMarket: true
        });

        const { asFragment } = render(<Catalog />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders null when no products are selected and not in no purchase market", () => {
        const { asFragment } = render(<Catalog />);
        expect(asFragment()).toMatchSnapshot();
    });
});
