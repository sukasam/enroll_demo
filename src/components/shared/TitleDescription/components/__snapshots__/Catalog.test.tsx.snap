// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders correctly with no purchase market 1`] = `<DocumentFragment />`;

exports[`Catalog renders correctly with selected UFG product 1`] = `
<DocumentFragment>
  <div
    class="css-1rmms3i-catalogContainer"
    data-testid="catalog_unimate_flavor"
  >
    <div>
      <span>
        catalog_unimate_flavor
      </span>
       Lemon
    </div>
    <div>
      <span>
        catalog_balance_flavor
      </span>
       Orange
    </div>
  </div>
</DocumentFragment>
`;

exports[`Catalog renders correctly with selected digital product 1`] = `
<DocumentFragment>
  <div
    data-testid="catalog_digital_product"
  >
    <span>
      catalog_basic_distributor_only
    </span>
  </div>
</DocumentFragment>
`;

exports[`Catalog renders null when no products are selected and not in no purchase market 1`] = `<DocumentFragment />`;
